bin/input_assure.py

```diff
@@ -0,0 +1,89 @@
+#!/usr/bin/env python
+
+import json
+import argparse
+import csv
+import gzip
+import sys
+
+
+def open_file(file_path, mode):
+    # Open a file based on the file extension
+    if file_path.endswith(".gz"):
+        return gzip.open(file_path, mode)
+    else:
+        return open(file_path, mode)
+
+
+def check_inputs(json_file, sample_id, address, output_error_file, output_json_file):
+    with open_file(json_file, "rt") as f:
+        json_data = json.load(f)
+
+    # Define a variable to store the match_status (True or False)
+    match_status = sample_id in json_data
+
+    # Initialize the error message
+    error_message = None
+
+    # Check for multiple keys in the JSON file and define error message
+    keys = list(json_data.keys())
+    original_key = keys[0] if keys else None
+
+    if len(keys) == 0:
+        error_message = f"{json_file} is completely empty!"
+        print(error_message)
+        sys.exit(1)
+    elif len(keys) > 1:
+        # Check if sample_id matches any key
+        if not match_status:
+            error_message = f"No key in the MLST JSON file ({json_file}) matches the specified sample ID '{sample_id}'. The first key '{original_key}' has been forcefully changed to '{sample_id}' and all other keys have been removed."
+            # Retain only the specified sample ID
+            json_data = {sample_id: json_data.pop(original_key)}
```

Comment:
Curious as to why the behaviour is to coerce a key to match the `sample_id`. Is it that we assume people will attach MLST JSON files with keys that are not `sample_id`? `input_assure.py` is not assuring a proper file is matched for the sample but more a proper MLST JSON has been linked? Don't know if it is a functionality that'd be used but maybe this behaviour could be a parameter within the script, and thus toggled in nextflow process?